patches/server/0657-Missing-Entity-Behavior-API.patch

```diff
@@ -368,6 +482,140 @@ index 7adda5c93e7c172ea0ba4a3f15828b5e54a283e7..fffaf4108b632ceabac4186d93b34ad0
 +    }
 +    // Paper end
  }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
+index ae05f526f9ec70a2992ef3ee66b7f57eca2351fc..15e2c2653d7d41e8bafa0ffe1afaa733c569b2ab 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLlama.java
+@@ -64,4 +64,36 @@ public class CraftLlama extends CraftChestedHorse implements Llama, com.destroys
+     public EntityType getType() {
+         return EntityType.LLAMA;
+     }
++
++    // Paper start
++    @Override
++    public boolean inCaravan() {
++        return this.getHandle().inCaravan();
++    }
++
++    @Override
++    public void joinCaravan(@org.jetbrains.annotations.NotNull Llama llama) {
++        this.getHandle().joinCaravan(((CraftLlama) llama).getHandle());
```

Comment:
I believe we would want a precondition check here (or a Validate, idk what the current flavour is we want). The annotation itself won't cut it at runtime.

Reply:
For a readable and understandable null pointer exception.
If someone passes null there, they get an NPE saying  "get handle cannot be called because llama is null"  or along the lines of that.

Reply:
This is a very similar PR which has something like this. https://github.com/PaperMC/Paper/pull/7089/files

Imo, null pointer should explain enough? But if others think that’s fine ig?